lib/comgr/src/comgr-compiler.cpp

```diff
@@ -581,6 +582,23 @@ InProcessDriver::InProcessDriver(raw_ostream &DiagOS)
 }
 
 amd_comgr_status_t InProcessDriver::Execute(ArrayRef<const char *> Args) {
+  // For HIP compilation, we launch a process.
+  bool IsHIP = false;
+  for (auto &Arg : Args) {
+    if (StringRef(Arg) == "--genco") {
+      IsHIP = true;
+    }
+  }
+  if (IsHIP) {
+    std::stringstream SS;
+    SS << "hipcc ";
+    for (auto &Arg : Args) {
+      SS << " " << Arg;
+    }
+    return system(SS.str().c_str()) == -1 ?
```

Comment:
I believe this will be the first out-of-process action in the library. I understand the requirements may differ for tools which use Comgr to compile HIP, but I think we should consider if this is possible in-process.

Reply:
The plan is to implement this in stages. In the first stage we will just launch hipcc as a process. In the future we will make it in-process.